### DISCUSSION TITLE
Specific dependency versions in package.json

### DISCUSSION THREAD

**Discussion by eppsilon** [nrwl/nx#7756] [General]
Why does NX install specific versions of NPM packages instead of a semver range? I wasn't able to find anything in the docs explaining it.

**Comment by leon** [comment]
I think it's because they lock specific nx versions to specific dependencies.
that way they can ensure that nx 13.0.0 works with the complete set of dependencies.

otherwise if you updated from x.2.0 to x.3.0 of a dependency it might break everything.

then when you run `nx migrate latest` it will update to the latest tested complete set of deps.

you can of course choose to update them yourself, but at your own risk.
